### PR TITLE
based64: fix a bug and make it more testable

### PR DESCRIPTION
**Pull request by rminnich**
Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
fixes #2498

File: cmds/core/base64/base64.go [hunk 1]
```diff
@@ -34,31 +38,39 @@ func do(r io.Reader, w io.Writer, decode bool) error {
...
+// makes testing a tad easier (so we don't have an if in main()
+// allows us, should we wish, in future, to go with using
+// names[1] as out. base64 commands are very nonstandard.
+func run(stdin io.Reader, stdout io.Writer, decode bool, names ...string) error {
+	// fbsd lets you name an output file,
+	switch len(names) {
+	case 0:
+	case 1:
+		name := names[0]
+		if name == "-" {
```
Comment (10000TB) [inline cmds/core/base64/base64.go:56]: nit: base64 in mac also recognies two hyphens 

```
➜  ~ echo hahah | base64
aGFoYWgK
➜  ~ echo hahah | base64 -
aGFoYWgK
➜  ~ echo hahah | base64 --
aGFoYWgK
➜  ~ echo hahah | base64 ---
base64: unrecognized option `---'
Usage:	base64 [-hvDd] [-b num] [-i in_file] [-o out_file]
  -h, --help     display this message
  -Dd, --decode   decodes input
  -b, --break    break encoded string into num character lines
  -i, --input    input file (default: "-" for stdin)
  -o, --output   output file (default: "-" for stdout)
➜  ~
```

Comment (rminnich) [reply cmds/core/base64/base64.go:56]: that's horrible :-)
one is bad enough :-)
I am going to remove - support entirely. If people want a placeholder like that, they can use
/dev/stdin
works on osx and linux ...

Comment (10000TB) [reply cmds/core/base64/base64.go:56]: ok, looks like that is GNU way. GNU binary follows same

```
xuehaohu@worldpeace:~$ echo  hhh | base64
aGhoCg==
xuehaohu@worldpeace:~$ echo  hhh | base64 -
aGhoCg==
xuehaohu@worldpeace:~$ echo  hhh | base64 --
aGhoCg==
xuehaohu@worldpeace:~$ echo  hhh | base64 ---
base64: unrecognized option '---'
Try 'base64 --help' for more information.
```

Comment (rminnich) [reply cmds/core/base64/base64.go:56]: I am going to stand my ground and refuse to continue this -- nonsense, and see how many arrows land in my back.

File: cmds/core/base64/base64.go [hunk 2]
```diff
@@ -15,14 +15,18 @@ package main
...
 	"fmt"
 	"io"
 	"log"
 	"os"
 )
 
-var decode = flag.Bool("d", false, "Decode")
+var (
+	decode      = flag.Bool("d", false, "Decode")
+	errBadUsage = errors.New("Usage: base64 [-d] [file]")
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/core/base64/base64.go]: thanks, I hope I got it right this time :-)

**Pull request merged**